### DISCUSSION TITLE
Godot-Jolt fork requirement

### DISCUSSION THREAD

**Discussion by smtucker** [grazianobolla/godot-multiplayer-template#17] [General]
Thank you for all your hard work on this project. It fills a nice gap in currently open high level network APIs for Godot, and I have been keeping a close eye on it.

The readme specifies that a specific fork of Godot-Jolt is required, but I didn't see anything specifically modified. May I please ask:

1) Is godot-monke-net already not compatible with the base Godot-Jolt project?
2) Why the need for a fork of Godot-Jolt?

**Comment by smtucker** [comment]
Never mind! I found the exposed stepping code in the 'stable' branch of your Godot-Jolt fork.

**Comment by grazianobolla** [comment]
Hi @smtucker, sorry for the late reply. Yes! The "custom code" would be the `space_step()` `space_flush_queries()` and `simulate()`

These exposed functions have no impact on the physics itself, but since Jolt doesn't expose them by default (nor does Godot PhysicsServer3D) I had to create this fork and expose them myself, they are needed for doing rollback in our case. I've also been in contact with Godot-Jolt maintainers and they have helped me troubleshoot some stuff!

Sadly without this methods it is impossible to implement real physics rollback, so yeah, you need to do some kind of modification to be able to step physics manually.

For the upcoming Godot release 4.4, Jolt will be integrated to the engine as a _module_ instead of a GDExtension, this means that for exposing these methods, I will have to modify the module code and re-compile the whole engine... Or in the best case scenario I can keep using the Godot-Jolt extension for a little while.

Robust client-server solutions are in its infancy for Godot (including this one), I've tried to provide a solution that works for me, but please feel welcome to give me any feedback!!
